### PR TITLE
[BPK-3839] add icon support to chips

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/Chip/Classes/BPKChip.m [hunk 1]
```diff
@@ -30,8 +30,9 @@
 @interface BPKChip ()
 @property(nonatomic) BPKLabel *titleLabel;
 @property(nonatomic) CALayer *tintLayer;
-
-@property(nonatomic) UIColor *textColor;
+@property(nonatomic) UIColor *contentColor;
+@property(nullable, nonatomic) UIView *iconView;
```
Comment (k0nserv) [inline Backpack/Chip/Classes/BPKChip.m:34]: Why isn't this a `BPKIconView` here?

File: Backpack/Chip/Classes/BPKChip.h [hunk 2]
```diff
@@ -77,6 +79,11 @@ IB_DESIGNABLE @interface BPKChip : UIControl
  */
 @property(nullable, nonatomic) UIColor *backgroundTint;
 
+/**
+ * An optional icon
+ */
+@property(nullable, nonatomic) BPKIconName iconName;
```
Comment (k0nserv) [inline Backpack/Chip/Classes/BPKChip.h:85]: This should be copy

Comment (georgegillams) [reply Backpack/Chip/Classes/BPKChip.h:85]: Fixed

File: Backpack/Chip/Classes/BPKChip.m (hunk 1 above)
Comment (georgegillams) [reply Backpack/Chip/Classes/BPKChip.m:34]: We don't do anything to it that a UIView can't do, so it seems unecessary to constrain it to a `BPKIconView`.

Comment (k0nserv) [reply Backpack/Chip/Classes/BPKChip.m:34]: But it is a `BPIconView`. It's not about what we do it, it's about helping the person reading the code understand. Reading `UIView` I started assuming that we might sometimes use a `BPKIconView` and other times some other type which hurts readability

Comment (georgegillams) [reply Backpack/Chip/Classes/BPKChip.m]: Fixed

**Pull request merged**